dlt/common/runtime/segment.py

```diff
@@ -170,7 +181,7 @@ def _send_event(event_name: str, properties: StrAny, context: StrAny) -> None:
         logger.debug("Skipping request to external service: payload was filtered out.")
         return
 
-    if not _WRITE_KEY:
+    if not _WRITE_KEY and _SEGMENT_ENDPOINT != RunConfiguration.dlthub_telemetry_endpoint:
```

Comment:
here you can simply check if _SEGMENT_ENDPOINT is None, if so, return. all the other checks where done in the init.